sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -182,4 +182,20 @@ object SQLExecution {
       body
     }(exec)
   }
+
+  /**
+   * Wrap passed function to ensure necessary thread-local variables like
+   * SparkContext local properties are forwarded to execution thread
+   */
+  def withThreadLocalCaptured[T](
```

Comment:
How about we make this also produce a (java) Future?

Reply:
I see previously it was using a Future which was moved to Callable in https://github.com/apache/spark/pull/24595

Reply:
It needed to be cancellable. So we are using java futures instead.

Reply:
I understand, will update to use java Futures instead

Reply:
@hvanhovell i have a suggestion, i see `org.apache.spark.FutureAction` as part of spark code which supports cancellation, do you think its a better approach instead.?

Reply:
@hvanhovell org.apache.spark.sql.execution.SubqueryExec#executeCollect needs now to await over java Future instead of scala future which would mean we have to have a new method under org.apache.spark.util.ThreadUtils#awaitResult, is this acceptable.?

Reply:
I think it's fine to have another `awaitResult`

Reply:
+1